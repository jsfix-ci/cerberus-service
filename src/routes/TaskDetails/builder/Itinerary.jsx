import React from 'react';

import { DATE_FORMATS, STRINGS } from '../../../utils/constants';
import { MovementUtil } from '../../../utils';

const Itinerary = ({ version }) => {
  const itinerary = MovementUtil.movementItinerary(MovementUtil.movementJourney(version));
  return (
    <div className="task-details-container">
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Itinerary</h3>
      <div className="bottom-border-thin" />
      {itinerary && itinerary.map((it, index) => {
        return (
          <div
            key={index}
            className={`${index !== itinerary.length - 1 && 'bottom-border-thin'} govuk-!-margin-top-1 govuk-!-padding-bottom-1`}
          >
            {index !== 0 && index <= itinerary.length - 1 && MovementUtil.itinRelativeTime(index, it, itinerary)}
            <div className="font__bold">
              {MovementUtil.itinFlightNumber(it)} <span className="dot__light" />&nbsp;
              {MovementUtil.departureLoc(it)} <span className="right-arrow">&#8594;</span>&nbsp;
              {MovementUtil.arrivalLoc(it)} <span className="dot__light" />&nbsp;
              {MovementUtil.formatDepartureTime(it, DATE_FORMATS.LONG_DAY_DATE)}
            </div>
            <div className="font__light">
              {MovementUtil.itinDepartureCountryCode(it)} <span className="right-arrow">&#8594;</span>&nbsp;
              {MovementUtil.itinArrivalCountryCode(it)} <span className="dot__light" />&nbsp;
              {STRINGS.ARRIVAL_TEXT} {MovementUtil.formatArrivalTime(it, DATE_FORMATS.LONG)}
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default Itinerary;
