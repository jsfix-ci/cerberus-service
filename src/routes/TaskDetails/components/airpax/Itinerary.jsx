import React from 'react';

import { DATE_FORMATS, STRINGS } from '../../../../utils/constants';
import { JourneyUtil, MovementUtil } from '../../../../utils';

const Itinerary = ({ version }) => {
  const itinerary = JourneyUtil.movementItinerary(JourneyUtil.get(version));
  return (
    <div className="task-details-container">
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Itinerary</h3>
      <div className="bottom-border-thin" />
      {itinerary && itinerary.map((it, index) => {
        return (
          <div
            key={index}
            className={`${index !== itinerary.length - 1 && 'bottom-border-thin'} govuk-!-margin-top-1 govuk-!-padding-bottom-1`}
          >
            {index !== 0 && index <= itinerary.length - 1 && MovementUtil.itinRelativeTime(index, it, itinerary)}
            <div className="font__bold">
              {MovementUtil.itinFlightNumber(it)} <span className="dot__light" />&nbsp;
              {JourneyUtil.departureLoc(it)} <span className="right-arrow">&#8594;</span>&nbsp;
              {JourneyUtil.arrivalLoc(it)} <span className="dot__light" />&nbsp;
              {JourneyUtil.formatDepartureTime(it, DATE_FORMATS.LONG_DAY_DATE)}
            </div>
            <div className="font__light">
              {MovementUtil.itinDepartureCountryCode(it)} <span className="right-arrow">&#8594;</span>&nbsp;
              {MovementUtil.itinArrivalCountryCode(it)} <span className="dot__light" />&nbsp;
              {STRINGS.ARRIVAL_TEXT} {JourneyUtil.formatArrivalTime(it, DATE_FORMATS.LONG)}
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default Itinerary;
