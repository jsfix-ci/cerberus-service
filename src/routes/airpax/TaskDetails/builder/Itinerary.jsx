import React from 'react';

import { LONG_DAY_DATE_FORMAT, LONG_DATE_FORMAT, ARRIVAL_TEXT } from '../../../../constants';
import { MovementUtil } from '../../utils';

const Itinerary = ({ version }) => {
  const itinerary = MovementUtil.movementItinerary(MovementUtil.movementJourney(version));
  return (
    <div className="task-details-container">
      <h3 className="govuk-heading-m govuk-!-margin-top-0">Itinerary</h3>
      <div className="bottom-border-thin" />
      {itinerary && itinerary.map((it, index) => {
        return (
          <div
            key={index}
            className={`${index !== itinerary.length - 1 && 'bottom-border-thin'} govuk-!-margin-top-1 govuk-!-padding-bottom-1`}
          >
            {index !== 0 && index <= itinerary.length - 1 && MovementUtil.itinRelativeTime(index, it, itinerary)}
            <div className="font__bold">
              {MovementUtil.itinFlightNumber(it)} <span className="dot__light" />&nbsp;
              {MovementUtil.departureLoc(it)} <span className="right-arrow">&#8594;</span>&nbsp;
              {MovementUtil.arrivalLoc(it)} <span className="dot__light" />&nbsp;
              {MovementUtil.formatDepartureTime(it, LONG_DAY_DATE_FORMAT)}
            </div>
            <div className="font__light">
              {MovementUtil.itinDepartureCountryCode(it)} <span className="right-arrow">&#8594;</span>&nbsp;
              {MovementUtil.itinArrivalCountryCode(it)} <span className="dot__light" />&nbsp;
              {ARRIVAL_TEXT} {MovementUtil.formatArrivalTime(it, LONG_DATE_FORMAT)}
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default Itinerary;
