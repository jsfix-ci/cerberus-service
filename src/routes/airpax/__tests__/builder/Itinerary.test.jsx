import React from 'react';
import renderer from 'react-test-renderer';

import Itinerary from '../../TaskDetails/builder/Itinerary';
import taskDetailsData from '../../__fixtures__/taskData_AirPax_TaskDetails.fixture.json';

describe('Itinerary', () => {
  it('should render the itinerary component', () => {
    const tree = renderer.create(<Itinerary version={taskDetailsData.versions[1]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
