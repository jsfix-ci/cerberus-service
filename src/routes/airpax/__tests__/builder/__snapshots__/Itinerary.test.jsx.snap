// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary should render the itinerary component 1`] = `
<div
  className="task-details-container"
>
  <h3
    className="govuk-heading-m govuk-!-margin-top-0"
  >
    Itinerary
  </h3>
  <div
    className="bottom-border-thin"
  />
  <div
    className="bottom-border-thin govuk-!-margin-top-1 govuk-!-padding-bottom-1"
  >
    <div
      className="font__bold"
    >
      AC0850
       
      <span
        className="dot__light"
      />
       
      YYC
       
      <span
        className="right-arrow"
      >
        →
      </span>
       
      LHR
       
      <span
        className="dot__light"
      />
       
      Wed 19 Sep 2018 at 18:25
    </div>
    <div
      className="font__light"
    >
      CA
       
      <span
        className="right-arrow"
      >
        →
      </span>
       
      GB
       
      <span
        className="dot__light"
      />
       
      Arrival
       
      19 Sep 2018 at 10:00
    </div>
  </div>
  <div
    className="bottom-border-thin govuk-!-margin-top-1 govuk-!-padding-bottom-1"
  >
    <div
      className="font__light"
    >
      14 days later
    </div>
    <div
      className="font__bold"
    >
      AC0850
       
      <span
        className="dot__light"
      />
       
      CDG
       
      <span
        className="right-arrow"
      >
        →
      </span>
       
      YYZ
       
      <span
        className="dot__light"
      />
       
      Wed 3 Oct 2018 at 11:00
    </div>
    <div
      className="font__light"
    >
      FR
       
      <span
        className="right-arrow"
      >
        →
      </span>
       
      CA
       
      <span
        className="dot__light"
      />
       
      Arrival
       
      3 Oct 2018 at 13:05
    </div>
  </div>
  <div
    className="false govuk-!-margin-top-1 govuk-!-padding-bottom-1"
  >
    <div
      className="font__light"
    >
      3 hours later
    </div>
    <div
      className="font__bold"
    >
      AC0850
       
      <span
        className="dot__light"
      />
       
      YYZ
       
      <span
        className="right-arrow"
      >
        →
      </span>
       
      YYC
       
      <span
        className="dot__light"
      />
       
      Wed 3 Oct 2018 at 16:05
    </div>
    <div
      className="font__light"
    >
      CA
       
      <span
        className="right-arrow"
      >
        →
      </span>
       
      CA
       
      <span
        className="dot__light"
      />
       
      Arrival
       
      3 Oct 2018 at 18:16
    </div>
  </div>
</div>
`;
